src/bdc/steps/regionalatlas.py

```diff
@@ -230,17 +229,22 @@ def calculate_regional_score(self, lead) -> float | None:
         :return: float | None - The computed score if the necessary fields are present for the lead. None otherwise.
         """
 
-        if (
-            lead[f"{self.name.lower()}_pop_density"] is None
-            or lead[f"{self.name.lower()}_employment_rate"] is None
-            or lead[f"{self.name.lower()}_disp_income_p_inhabitant"] is None
-        ):
-            return None
+        pop_density_col = f"{self.name.lower()}_pop_density"
+        employment_rate_col = f"{self.name.lower()}_employment_rate"
+        income_col = f"{self.name.lower()}_disp_income_p_inhabitant"
+
+        pop_density = lead[pop_density_col]
+        employment_rate = lead[employment_rate_col]
+        income_per_inhabitant = lead[income_col]
+
+        pop_density = pop_density if pd.notnull(pop_density) else 0
+        employment_rate = employment_rate if pd.notnull(employment_rate) else 0
+        income_per_inhabitant = (
+            income_per_inhabitant if pd.notnull(income_per_inhabitant) else 0
+        )
 
         regional_score = (
-            lead[f"{self.name.lower()}_pop_density"]
-            * lead[f"{self.name.lower()}_employment_rate"]
-            * lead[f"{self.name.lower()}_disp_income_p_inhabitant"]
+            pop_density * employment_rate * income_per_inhabitant
         ) / 1000000
 
-        return regional_score
+        return regional_score if pd.notnull(regional_score) else None
```

Comment:
Would this still possibly return None? Maybe if it is None here it is better to throw an exception?

Reply:
Agree, it can't possibly be None but I wanted to fortify twofold it. Exception is a better idea. Addressed